src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerResponseToServiceApplication.java

```diff
@@ -90,13 +91,39 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
         return AboutToStartOrSubmitResponse.<CaseData, State>builder().build();
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore
+    ) {
+        CaseData caseData = details.getData();
+        List<String> errors = new ArrayList<>();
+
+        AlternativeService altService = caseData.getAlternativeService();
+        if (isNotEmpty(altService.getServiceApplicationDocuments())
+            && altService.getServiceApplicationDocuments().size() > MAX_ATTACHMENTS_PER_APP) {
+            errors.add("Exceeds allowed number of document attachments");
```

Comment:
Might be useful to include the max number of attachments in this error for the caseworker